script/tool/lib/src/publish_check_command.dart

```diff
@@ -63,6 +64,33 @@ class PublishCheckCommand extends PluginCommand {
     }
   }
 
+  Future<bool> hasValidPublishCheckRun(Directory package) async {
+    final io.Process process = await io.Process.start(
+      'flutter',
+      <String>['pub', 'publish', '--', '--dry-run'],
+      workingDirectory: package.path,
+    );
+
+    String output = '';
+    process.stdout.listen((List<int> event) {
+      io.stdout.add(event);
+      output += String.fromCharCodes(event);
+    });
+
+    process.stderr.listen((List<int> event) {
+      io.stderr.add(event);
+      output += String.fromCharCodes(event);
+    });
+
+    final int exitCode = await process.exitCode;
+
+    final bool hasPreReleaseSdk = output.contains('Package has 1 warning.') &&
```

Comment:
Aren't errors and warnings reported separately? What does the output for a package with, say, 1 warning and 1 error look like? We need to make sure this won't pass a packages in that case.

Reply:
From https://github.com/dart-lang/pub/blob/master/lib/src/command/lish.dart#L176 it looks like maybe we need to ensure that it doesn't contain "can't be published".

Reply:
Oh wait, it early-returns after errors, before getting to warnings. So apparently we can't get the warning output if it's actually failed.